### ISSUE TITLE
A Failing Pre-Install hooks should block chart install 

### ISSUE THREAD

**Issue by endzyme** [reactiveops/reckoner#76]
Currently the behavior of reckoner is to continue on if any of the pre or post installation hooks fail. This can be a confusing user experience, especially if you've fired 20 chart installs and the first one had a failure. This signal can be lost in the noise of output from Reckoner. 

Feature Request
- Could Reckoner, by default, stop after the first failed pre-install hook and notify the customer.
- Could we add a quick summary of failed chart installations at the end of the run (failed chart install N shouldn't block chart N+1)
- Could we add an override to ignore all failed pre-install-hooks
- Failed post-install hooks should also be added to the summary list of failed chart installs as "Failed  Post-Install"
- All hooks should stop running after the first one fails (example below)

```yaml
charts:
  my-chart:
    set-values:
      enabled: true
    hooks:
      pre_install:
        - echo true
        - curl non*possible().domain
        - echo true   #<--- this should never fire if the curl above fails & chart will also not be installed
      post_install:
        - echo true
        - cat fileno-existy
        - echo this-never-runs-in-post
```

**Comment by ejether** [comment]
Ah, I think the change @sudermanjr created was that if the hook fails, the chart won't be installed.
This is so that if any hook command fails, the rest of the hooks should not be run.

**Comment by endzyme** [comment]
Do you have that PR handy? Was this in the previous version before the rename? I am curious if I clobbered that behavior accidentally with my changes recently. I think this behavior should have a test for sure and I'll see if I can make some time to test it soon.

**Comment by ejether** [comment]
I'll see if I can dig it up. I'm 99% sure we had it exit on failure of a hook.

**Comment by endzyme** [comment]
Yup ok - so I traced back behavior on and old ref. 

1. Calls install on a chart definition https://github.com/reactiveops/reckoner/blob/307e66b551386e82406d84f90449a3908caed2ef/reckoner/course.py#L119 
2. Install runs pre_install hook https://github.com/reactiveops/reckoner/blob/307e66b551386e82406d84f90449a3908caed2ef/reckoner/chart.py#L194
3. run_hook runs the command and raises an error if exitcode > 0 (https://github.com/reactiveops/reckoner/blob/307e66b551386e82406d84f90449a3908caed2ef/reckoner/chart.py#L150-L153 & https://github.com/reactiveops/reckoner/blob/307e66b551386e82406d84f90449a3908caed2ef/reckoner/__init__.py#L84-L85)
4. That raised error pops out of the chart.install() and bubbles to course iteration of install on charts https://github.com/reactiveops/reckoner/blob/307e66b551386e82406d84f90449a3908caed2ef/reckoner/course.py#L120-L126 which is supposed to catalog the error and continue on

Looks like my refactor for testing and other things missed this component. I can take this on to fix it since I introduced the bad behavior!

**Comment by ejether** [comment]
Cool, I was just tracing that as well. I agree with your assessment. 
Also, I'm not sure this chunk of code is going to get run in an error since the `raise` should cause prevent the further execution of the method right?
https://github.com/reactiveops/reckoner/blob/master/reckoner/chart.py#L163

**Comment by endzyme** [comment]
RE: Your last note - we are only catching exceptions if the call python function fails. I changed the behavior of `call()` to always respond with a `Respond` class. It's up to the caller to determine if exitcode should raise an error.

**Comment by ejether** [comment]
Ah, so that is the root of the behavior change then?

**Comment by endzyme** [comment]
Yup exactly - that's the core of what introduced the bug

**Comment by endzyme** [comment]
Fixed with #77
